core/src/main/java/kafka/server/share/DelayedShareFetch.java

```diff
@@ -248,14 +250,18 @@ private void processAcquiredTopicPartitionsForLocalLogFetch(LinkedHashMap<TopicI
                 // updated in a different tryComplete thread.
                 responseData = combineLogReadResponse(topicPartitionData, localPartitionsAlreadyFetched);
 
+            resetFetchOffsetMetadataForRemoteFetchPartitions(topicPartitionData, responseData);
+
             List<ShareFetchPartitionData> shareFetchPartitionDataList = new ArrayList<>();
-            responseData.forEach((topicIdPartition, logReadResult) ->
-                shareFetchPartitionDataList.add(new ShareFetchPartitionData(
-                    topicIdPartition,
-                    topicPartitionData.get(topicIdPartition),
-                    logReadResult.toFetchPartitionData(false)
-                ))
-            );
+            responseData.forEach((topicIdPartition, logReadResult) -> {
+                if (logReadResult.info().delayedRemoteStorageFetch.isEmpty()) {
```

Comment:
Question: What happens if we have some `delayedRemoteStorageFetch` data in logReadResut now? Is it ignored or further processed?